### DISCUSSION TITLE
How to Customise Password Pusher

### DISCUSSION THREAD

**Discussion by ZuluPilgrim** [pglombardo/PasswordPusher#950] [Q&A] [answered]
I have decided to try the Docker Route using the ephemeral version. It has installed fine and runs as expected. 

I now wish to customise the install using /opt/PasswordPusher/config/settings.yml. When I however search, I see two version so I am not sure which to edit.

- /var/lib/docker/overlay2/347d353d6a4f808a0f43117cd890b8a3d71411261824a0682e9066b1eae2a043/merged/opt/PasswordPusher/config/settings.yml
- /var/lib/docker/overlay2/bd0f9d909c95f388cc5a32d4a4abb3ab7c5fe7e05f54d444a09b93d453051c3f/diff/opt/PasswordPusher/config/settings.yml

**Comment by pglombardo** [comment]
Hi @ZuluPilgrim - there are instructions on how to apply a custom settings.yml file here:
https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#configuring-via-a-custom-settingsyml-file

The point being is that you have an external version that is mounted inside of the container on boot.  This should be what you're looking for.  Let me know!

**Comment by ZuluPilgrim** [comment]
@pglombardo sorry about all the questions; I am in noob land here. 

I saw the above documentation but was not understanding some of it. A slower, more careful read saw me come right :)

If I wish to edit the layout of the served webpage, I am guessing that I then edit some files found in target=/opt/PasswordPusher/ ?

Many thanks again

**Reply by pglombardo** [reply]
No problem at all with the questions!  What is it you want to do?  

Change the theme or rebrand the site?  If so, we have easy options for that:

1. How to [change the theme](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#themes)
2. How to [re-brand the site](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#rebranding)
3. If you want to add custom CSS, [see here](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#how-to-add-custom-css).

Hopefully one of those will cover what you want to do.  If not, let me know!

**Reply by ZuluPilgrim** [reply]
@pglombardo , For the logos eg PWP__BRAND__LIGHT_LOGO, I understand how to update the settings.yml file.  but I am not sure about is where I same the actual images. Looking at my current installed the default images are stored in both the diff and merged folders. Do I need to upload copies of the new logo there and then update the settings file accordingly?

**Reply by pglombardo** [reply]
I added an explanation and [better instructions here](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#rebranding).  This should help you get images into the container if needed.

**Reply by ZuluPilgrim** [reply]
Thanks

**Reply by ZuluPilgrim** [reply]
Here is my docker command 

docker run -d --env PWPUSH_MASTER_KEY=1234567891011 --mount type=bind, source=/home/sysadmin/pwdpusher/settings.yml,target=/opt/PasswordPusher/config/settings.yml -p "5100:5100" -v /home/sysadmin/pwdpusher/logos:/opt/PasswordPusher/public/logos pglombardo/pwpush-ephemeral:release

The settings work.

I have uploaded my logo called padlock.png to /home/sysadmin/pwdpusher/logos and added light_logo: 'padlock.png' to settings.yml

When restarting the container, my logos are broken links i.e. Failed to load resource: the server responded with a status of 404 (Not Found)

**Reply by pglombardo** [reply]
What is the value of `PWP__BRAND__LIGHT_LOGO`?

**Reply by ZuluPilgrim** [reply]
Because I am using settings.yml I only set light_logo:

 ```
### Site logo
  #
  # ..for both a light and dark theme
  #
  # You can also replace these relative paths with fully qualified HTTP links to
  # external resources such as Amazon S3 etc.
  # e.g. PWP__BRAND__DARK_LOGO='https://mys3bucket.amazonaws.com/a/some-image.png'
  #
  # Environment variable override: PWP__BRAND__LIGHT_LOGO='https://mys3bucket.amazonaws.com/a/lea+giuliana.png'
  # Environment variable override: PWP__BRAND__DARK_LOGO='https://mys3bucket.amazonaws.com/a/lea+giuliana.png'
  #
  # light_logo: 'logo-transparent-sm-bare.png'
  # dark_logo: 'logo-transparent-sm-dark-bare.png'
  light_logo: 'padlock.png'

  ### Icon images for mobile.  When people on mobile (phones/tablets), bookmark
  # the site or it is shown in history, these icons are used.
  #
  # You can also replace these relative paths with fully qualified HTTP links to
  # external resources such as Amazon S3 etc.
  # e.g. PWP__BRAND__ICON_57x57='https://mys3bucket.amazonaws.com/a/some-image.png'
  #
  # Environment variable override: PWP__BRAND__ICON_57x57='/path/to/image'
  # icon_57x57: 'apple-icon-57x57.png'
  # Environment variable override: PWP__BRAND__ICON_60x60='/path/to/image'
  # icon_60x60: 'apple-icon-60x60.png'
  # Environment variable override: PWP__BRAND__ICON_72x72='/path/to/image'
  # icon_72x72: 'apple-icon-72x72.png'
  # Environment variable override: PWP__BRAND__ICON_76x76='/path/to/image'
  # icon_76x76: 'apple-icon-76x76.png'
  # Environment variable override: PWP__BRAND__ICON_114x114='/path/to/image'
  # icon_114x
```

**Reply by pglombardo** [reply]
Try this:

```yaml
light_logo: '/logos/padlock.png'
```

**Reply by ZuluPilgrim** [reply]
Not sure how I missed that but did. Many thanks, it worked :)

**Comment by ZuluPilgrim** [comment, marked as answer]
@pglombardo perfect, this will do it. Many thanks

**Comment by ZuluPilgrim** [comment]
If I make a change in settings, do I need to restart the container to apply these?

**Reply by pglombardo** [reply]
Yes - changes are only picked up on boot.  If you changed the theme, make sure you set the `precompile` option to `true` too.

**Reply by ZuluPilgrim** [reply]
Perfect :)

**Comment by ZuluPilgrim** [comment]
Has something changed? I have just done a fresh install using the following:
```
sudo docker run -d \
  --env PWPUSH_MASTER_KEY=xxxxxxxxx \
  --mount type=bind,source=/home/sysadmin/pwpusher/settings.yml,target=/opt/PasswordPusher/config/settings.yml \
  -p "5100:5100" \
  --restart unless-stopped \
  --name pwpusher \
  pglombardo/pwpush-ephemeral:release \
  -v /home/sysadmin/pwpusher/logos:/opt/PasswordPusher/public/logos
```

In setttings.yml I have  light_logo: '/logos/logo-tranparent-pwp-light.png'
![image](https://user-images.githubusercontent.com/111164245/233757485-a1c83e98-9978-412d-bdb6-dae4fb0f01dd.png)

But I get
![image](https://user-images.githubusercontent.com/111164245/233757654-9fbe8360-79ac-4d2c-af6e-e3175dda5024.png)

**Reply by pglombardo** [reply]
Hi @ZuluPilgrim - Nothing has changed here with the `release` tag...
